File: Product of Array Except Self/Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answers = [1] * len(nums)
        
        temp = 1
        for i in range(len(nums)):
            answers[i] = temp
            temp *= nums[i]
        
        temp = 1
        for i in range(len(nums) -1, -1, -1):
            answers[i] *= temp
            temp *= nums[i]
        
        return answers

File: Product of Array Except Self/Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> productExceptSelf(nums.size());
        
        int temp = 1;
        for (int i = 0; i < nums.size(); ++i) {
            productExceptSelf[i] = temp;
            temp *= nums[i];
        }
        
        temp = 1;
        for (int i = nums.size() - 1; i >= 0; --i) {
            productExceptSelf[i] *= temp;
            temp *= nums[i]; 
        }
        
        return productExceptSelf;
    }
};